spec/forms/recovery_code_form_spec.rb

```diff
@@ -7,13 +7,13 @@
         user = create(:user)
         raw_code = RecoveryCodeGenerator.new(user).create
 
-        result = RecoveryCodeForm.new(user, raw_code).submit
+        form = RecoveryCodeForm.new(user, raw_code)
+        result = instance_double(FormResponse)
+        extra = { multi_factor_auth_method: 'recovery code' }
 
-        result_hash = {
-          success: true,
-        }
-
-        expect(result).to eq result_hash
+        expect(FormResponse).to receive(:new).
+          with(success: true, errors: {}, extra: extra).and_return(result)
+        expect(form.submit).to eq result
```

Comment:
I think we need to update the spec description for this test, which reads "it 'returns true for success?' do" currently